source/funkin/editors/character/CharacterEditor.hx

```diff
@@ -0,0 +1,696 @@
+					},
+					{
+						label: "Save As...",
+						keybind: [CONTROL, SHIFT, S],
+						onSelect: _file_saveas,
+					},
+					null,
+					{
+						label: "Exit",
+						keybind: [ESCAPE],
+						onSelect: _file_exit
+					}
+				]
+			},
+			{
+				label: "Edit",
+				childs: [
+					{
+						label: "Camera Position",
+						onSelect: moveCameraPos
+					},
+					{
+						label: "Properties",
+						onSelect: edit_properties_window,
+					}
+				]
+			},
+			{
+				label: "Animation",
+				childs: [
+					{
+						label: "New Animation",
+						keybind: [CONTROL, N],
+						onSelect: add_anim_window,
+					},
+					null,
+					{
+						label: "Update Animation",
+						keybind: [CONTROL, U],
+						onSelect: update_anim_window,
+					},
+					null,
+					{
+						label: "Remove Animation",
+						keybind: [CONTROL, D],
+						onSelect: remove_anim_window,
+					},
+				]
+			},
+			{
+				label: "Help",
+				childs: [
+					{
+						label: "Controls",
+						onSelect: help_controls_window
+					}
+				]
+			}
+		];
+
+		topMenuSpr = new UITopMenu(topMenu);
+		topMenuSpr.cameras = uiGroup.cameras = [uiCamera];
+
+		Framerate.fpsCounter.alpha = 0.4;
+		Framerate.memoryCounter.alpha = 0.4;
+		Framerate.codenameBuildField.alpha = 0.4;
+	}
+
+	var ofs_cross:FlxSpriteGroup = new FlxSpriteGroup(0, 0);
+
+	public function loadCross()
+	{
+		var i:Int = ofs_cross.members.length - 1;
+		while (i >= 0)
+		{
+			var memb:FlxSprite = ofs_cross.members[i];
+			if (memb != null)
+			{
+				memb.kill();
+				ofs_cross.remove(memb);
+				memb.destroy();
+			}
+			--i;
+		}
+		ofs_cross.clear();
+
+		var horizontal_cross = new FlxSprite(0, 0).makeGraphic(2000, 5);
+		var vertical_cross = new FlxSprite(0, 0).makeGraphic(5, 2000);
+		horizontal_cross.y += (vertical_cross.height / 2);
+		vertical_cross.x += (horizontal_cross.width / 2);
+		ofs_cross.add(horizontal_cross);
+		ofs_cross.add(vertical_cross);
+		horizontal_cross.scrollFactor.set(1, 1);
+		vertical_cross.scrollFactor.set(1, 1);
+		// ofs_cross.screenCenter();
+		ofs_cross.y = -200;
+		ofs_cross.scrollFactor.set(1, 1);
+		ofs_cross.alpha = 0.6;
+	}
+
+	var OFFSET_X:Float = 150;
+
+	public function reloadStage()
+	{
+		var i:Int = stage.members.length - 1;
+		while (i >= 0)
+		{
+			var memb:FunkinSprite = stage.members[i];
+			if (memb != null)
+			{
+				memb.kill();
+				stage.remove(memb);
+				memb.destroy();
+			}
+			--i;
+		}
+		stage.clear();
+		var playerXDifference = 0;
+		if (char.isPlayer)
+			playerXDifference = 670;
+
+		var bg = new FunkinSprite(-600 + OFFSET_X - playerXDifference, -200, Paths.image("stages/default/stageback"));
+		stage.add(bg);
+		bg.scrollFactor.set(0.9, 0.9);
+
+		var stageFront = new FunkinSprite(-600 + OFFSET_X - playerXDifference, 600, Paths.image("stages/default/stagefront"));
+		stageFront.setGraphicSize(Std.int(stageFront.width * 1.1));
+		stageFront.updateHitbox();
+		stage.add(stageFront);
+		stageFront.scrollFactor.set(0.9, 0.9);
+
+		stage.setPosition(0, 0);
```

Comment:
This code feels like its mostly from Psych

Reply:
Not quite, the variables just look identical. I made everything using the ecosystem of this engine and only use elements that has been made for this engine. There are some sections of code that remind the one and the other of psych engine, but it's all hand-written by me and every line of code I've ever written got tested by me.

Reply:
Are you able to switch between player side and opponent side?

Reply:
Yes.

Reply:
Can you record a video of that too?

Reply:
https://github.com/FNF-CNE-Devs/CodenameEngine/assets/75845687/52698534-ed92-4b1f-aa0e-1de54be3757a